Ansible 3

pavelvizir opened the pull request:
# Выполнено ДЗ №10

 - [x] Основное ДЗ
 - [x] Задание со *

## В процессе сделано:
 - Созданы ansible роли:
   - *db*
   - *app*
 - Созданы ansible окружения:
   - *stage*
   - *prod*
 - Практика с ansible vault:
   - *credentials.yml*
 - Практика с ansible community roles:
   - *jdauphant.nginx*
 - Static -> dynamic inventory.
 - TravisCI tests:
   - packer validate
   - terraform validate, tflint
   - ansible-lint

## Как запустить проект:
 - cd terraform/stage && terraform apply -auto-approve &&\
    cd ../../ansible && ansible-playbook playbooks/site.yml

## Как проверить работоспособность:
 - cd terraform/stage && firefox http://$(terraform output app_external_ip)

## PR checklist
 - [x] Выставил label с номером домашнего задания
 - [x] Выставил label с темой домашнего задания

pavelvizir committed:
Homework's 'ansible-3' main task completed.

pavelvizir committed:
Homework's 'ansible-3' asterisk tasks completed.

pavelvizir committed:
Change last terraform script cd command.

pavelvizir committed:
Add --var-file to tflint.

pavelvizir committed:
Fix travis to test all playbooks and packer templates, update README

pavelvizir committed:
Fix previous PRs

Large non-atomic commit.

Contents:
- Fix ToC links in README.md
- (packer-base) Delete redundant variables from packer/variables.json.example
- (terraform-2) Make source_image_family 'null' in packer/{app,db}.json
- (terraform-2) Fix 'prefix' in terraform/stage/backend.tf
- (terraform-2) Remove obsolete variable 'disk_image' from
  terraform/{stage,prod}/terraform.tfvars.example
- (terraform-2) Provision 'db_internal_ip' to puma.service with template

pavelvizir committed:
Fix previous PRs, small fixes.

Large non-atomic commit.

Contents:
- Fix redundant packer test in .travys.yml
- Fix redundant packer test in README.md
- Fix ToC links in README.md again
- Update README.md, mention fixes to previous PRs
- (ansible-2) Rename app, db groups to tag_reddit-app, tag_reddit-db everywhere

pavelvizir committed:
Fix ToC link in README.md for the last time

Nklya commented on .travis.yml:
```diff
@@ -4,6 +4,19 @@ language: bash
+- touch ~/.ssh/appuser
+- touch ~/.ssh/appuser.pub
+script:
+- ansible-lint -v ansible/playbooks/*.yml
```
ansible-lint в запуске из корня репозитория не видит папку ролей, и поэтому не проверяет код в них.
Например в роли jdauphant.nginx есть проблемы с его точки зрения, а у тебя он их не нашел.

pavelvizir replied:
Устранил все неполадки. 

Интересует организационный момент. В будущем в подобных обстоятельствах (коммиты для уже принятого PR) следует дожидаться повторного ревью или мержить сразу?

Nklya replied:
Можно сразу мержить. Если уверен что понял в чем были проблемы и поправил их.
Если нет, можно спрашивать.

Если в базовом задании проблем нет, то PR обычно апрувится и даются комментарии по проблемам. А фиксить их или нет это уже на ваше усмотрение.

Nklya replied:
Я кстати в тестах тоже обхожу ругань ansible-lint на роль nginx, только немного по-другому https://github.com/express42/otus-homeworks/blob/2018-02/homeworks/ansible-3/controls/ansible.rb#L14

Nklya commented on .travis.yml:
```diff
@@ -4,6 +4,19 @@ language: bash
   bash
+install:
+- sudo pip install ansible
+- sudo pip install ansible-lint
```
Версии лучше фиксировать для воспроизводимых билдов

Nklya commented on ansible/old/gce.ini:
```diff
@@ -0,0 +1,76 @@
+# Copyright 2013 Google Inc.
```
Зачем его копировать в old, если он нормально переместился в папку окружения?

Nklya reviewed (approved):
Добрый день.

Павел, условия ДЗ выполнены, все реализовано отлично 👍

pavelvizir committed:
Fix problems according to teacher's comments

- (ansible/old) Remove redundant files
- (.travis.yml) Install specific versions of pip packages
- (.travis.yml) Fix ansible-lint to check roles
- (README.md)   Update readme with new .travis.yml listing

pavelvizir committed:
Fix ansible-lint check path
